hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -2956,14 +2956,15 @@ func machineTemplateBuilders(hcluster *hyperv1.HostedCluster, nodePool *hyperv1.
 			return nil
 		}
 	case hyperv1.AzurePlatform:
+		var err error
 		template = &capiazure.AzureMachineTemplate{}
 		mutateTemplate = func(object client.Object) error {
 			o, _ := object.(*capiazure.AzureMachineTemplate)
-			spec, err := azureMachineTemplateSpec(hcluster, nodePool, o.Spec)
+			machineTemplateSpec, err = azureMachineTemplateSpec(nodePool, o.Spec)
```

Comment:
The `mutateTemplate` function is defined but not immediately called, which means that `machineTemplateSpec` is left empty (its original value) by the time the AzureMachineTemplate’s name is computed.